chain/tendermint/src/runtime/generated.rs

```diff
@@ -0,0 +1,693 @@
+
+pub struct AscSignedMsgTypeEnum(pub(crate) AscEnum<AscSignedMsgType>);
+
+impl AscType for AscSignedMsgTypeEnum {
+    fn to_asc_bytes(&self) -> Result<Vec<u8>, DeterministicHostError> {
+        self.0.to_asc_bytes()
+    }
+
+    fn from_asc_bytes(
+        asc_obj: &[u8],
+        api_version: &Version,
+    ) -> Result<Self, DeterministicHostError> {
+        Ok(Self(AscEnum::from_asc_bytes(asc_obj, api_version)?))
+    }
+}
+
+impl AscIndexId for AscSignedMsgTypeEnum {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::TendermintSignedMsgTypeEnum;
+}
+
+#[repr(u32)]
+#[derive(AscType, Copy, Clone)]
+pub(crate) enum AscSignedMsgType {
+    SignedMsgTypeUnknown,
+    SignedMsgTypePrevote,
+    SignedMsgTypePrecommit,
+    SignedMsgTypeProposal,
+}
+
+impl AscValue for AscSignedMsgType {}
+
+impl Default for AscSignedMsgType {
+    fn default() -> Self {
+        Self::SignedMsgTypeUnknown
+    }
+}
+
+#[repr(u32)]
+#[derive(AscType, Copy, Clone)]
+pub(crate) enum AscBlockIDFlag {
+    BlockIdFlagUnknown,
+    BlockIdFlagAbsent,
+    BlockIdFlagCommit,
+    BlockIdFlagNil,
+}
+
+impl AscValue for AscBlockIDFlag {}
+
+impl Default for AscBlockIDFlag {
+    fn default() -> Self {
+        Self::BlockIdFlagUnknown
+    }
+}
+
+#[repr(C)]
+#[derive(AscType)]
+pub(crate) struct AscEventData {
+    pub event: AscPtr<AscEvent>,
+    pub block: AscPtr<AscEventBlock>,
+}
+
+impl AscIndexId for AscEventData {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::TendermintEventData;
+}
+
+#[repr(C)]
+#[derive(AscType)]
+pub(crate) struct AscEventList {
+    pub new_block: AscPtr<AscEventBlock>,
+    pub transaction: AscPtr<AscEventTxArray>,
+    pub validator_set_updates: AscPtr<AscEventValidatorSetUpdates>,
+}
+
+impl AscIndexId for AscEventList {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::TendermintEventList;
+}
+
+#[repr(C)]
+#[derive(AscType)]
+pub(crate) struct AscBlock {
+    pub header: AscPtr<AscHeader>,
+    pub data: AscPtr<AscData>,
+    pub evidence: AscPtr<AscEvidenceList>,
+    pub last_commit: AscPtr<AscCommit>,
+}
+
+impl AscIndexId for AscBlock {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::TendermintBlock;
+}
+
+#[repr(C)]
+#[derive(AscType)]
+pub(crate) struct AscBlockID {
+    pub hash: AscPtr<AscHash>,
+    pub part_set_header: AscPtr<AscPartSetHeader>,
+}
+
+impl AscIndexId for AscBlockID {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::TendermintBlockID;
+}
+
+#[repr(C)]
+#[derive(AscType)]
+pub(crate) struct AscBlockParams {
+    pub max_bytes: i64,
+    pub max_gas: AscGas,
+}
+
+impl AscIndexId for AscBlockParams {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::TendermintBlockParams;
+}
+
+#[repr(C)]
+#[derive(AscType)]
+pub(crate) struct AscCommit {
+    pub height: i64,
+    pub round: i32,
+    pub block_id: AscPtr<AscBlockID>,
+    pub signatures: AscPtr<AscCommitSigArray>,
+    pub _padding: u32,
```

Comment:
> Did you put the padding because you faced memory alignment issues when testing with Tendermint?

That's right. It would be great if we could get rid of this 👍